docs/extend/extend-zowe-zos/lifecycling-with-zwesvstc.md

```diff
@@ -0,0 +1,87 @@
+
+## Zowe extensions
+
+If you are providing an extension to Zowe that wishes to be lifecycled, you should provide your own directory containing your Zowe lifecycle scripts `start.sh`, and optinally `validate.sh` and `configure.sh`.
+
+The `instance.env` file in the `<INSTANCE_DIR>` used to launch Zowe contains the variable `EXTENDER_COMPNENTS` whose value is a semi-colon separated list of fully qualified directorys paths containing extender lifecycle scripts.
+
+***Example***
+Vendor MYVENDOR has a product MYAPP that installs into `/usr/lpp/myvendor/myapp`.  This product is a Zowe extension so they wish it to be started and stopped with Zowe and run as an address space under the `ZWESVSTC` in the Zowe USS shell.   
+
+The script `/usr/lpp/myvendor/myapp/zowe/validate.sh` checks that the environment has been configured correctly and the script `/usr/lpp/myvendor/myapp/zowe/start.sh` starts the vendor application.
+
+The installation documentation for MYAPP instructs the system programmer to update the `instance.env` file and update the `EXTENDER_COMPONENTS` variable to point to the fully qualified path of the directory containing the Zowe lifecycle scripts.
+
+```
+EXTENDER_COMPONENTS='/usr/lpp/myvendor/myapp/zowe'
+```
+
+When the Zowe instance is launched by running `<INSTANCE_DIR>/bin/zowe-start.sh` this will call the `/usr/lpp/myvendor/myapp/zowe/start.sh` and started task will create an address space under `ZWESVSTC` for the vendor component.  When the Zowe instance is stopped the address space will be termninated.
+
+## Sample extensions
+
+### Sample Zowe API extension
+
+The repository https://github.com/zowe/sample-node-api contains an exemplary sample of a Zowe extension with  a node server providing sample APIs for looking at cars in a dealership, see [sample-node-api](https://github.com/zowe/sample-node-api/blob/master/README.md).  The [configure.sh](https://github.com/zowe/sample-node-api/blob/master/bin/configure.sh) script statically registers the API into the API Mediation Layer as well as a tile that includes the swagger definitions into the API catalog.
+
+### Sample Zowe Desktop and API catalog extension
+
+The repository https://github.com/zowe/sample-trial-app contains an exemplary sampe of a Zowe extension with a node server providing a web page that gives a user interface to the APIs included with the API sample above.  The [configure.sh](https://github.com/zowe/sample-trial-app/blob/master/bin/configure.sh) installs a web page onto the Zowe desktop using the utility script `<RUNTIME_DIR>/bin/utils/zowe-install-iframe-plugin.sh`.  The [configure.sh](https://github.com/zowe/sample-trial-app/blob/master/bin/configure.sh) also installs a tile onto the API Mediation Layer's API Catalog.
```

Comment:
"exemplary sampe" -> "sample". I don't think we should call it exemplary as it doesn't have a validate script and some of it's interactions in configure and start are a bit wonky